src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double INCREASE_RATE = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[CAPACITY];
+    }
+
+    public void increaseArray() {
+        T[] arrayCopy = (T[]) new Object[(int) (array.length * INCREASE_RATE)];
+        System.arraycopy(array, 0, arrayCopy, 0, array.length);
+        array = arrayCopy;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() >= array.length) {
+            increaseArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
+            throw new ArrayListIndexOutOfBoundsException("Index passed to the method is invalid");
+        }
+        if (size() >= array.length) {
+            increaseArray();
+        }
+        if (index < size()) {
+            System.arraycopy(array, index, array, index + 1, size() - index);
+            array[index] = value;
+            size++;
+        }
+        if (index == size()) {
+            add(value);
+        }
 
```

Comment:
Better to move this part to the begin of method
```suggestion
```

Reply:
Tried to move it to the begin of the method, but method is failed

java.lang.AssertionError: Test failed! Size of array should be 1but it is 2 
Expected :1
Actual   :2